internal/zinc-core/src/main/scala/sbt/internal/inc/InvalidationProfiler.scala

```diff
@@ -0,0 +1,267 @@
+package sbt.internal.inc
+
+import java.io.File
+
+import xsbti.UseScope
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Defines a profiler interface that translates to the profiling protobuf schema.
+ *
+ * The protobuf schema has been mildly inspired from pprof:
+ * https://github.com/google/pprof/blob/master/proto/profile.proto
+ *
+ * A profiler interface should only be used by project, and not globally, as
+ * this class is not thread safe.
+ */
+abstract class InvalidationProfiler {
+  def profileRun: RunProfiler
+  def registerRun(run: zprof.ZincRun): Unit
+}
+
+object InvalidationProfiler {
+  final val empty: InvalidationProfiler = new InvalidationProfiler {
```

Comment:
From the implementation I guess this is being used when no profiling should be done?

Reply:
Yes